src/LayerPlan.cpp

```diff
@@ -849,8 +849,13 @@ void LayerPlan::writeGCode(GCodeExport& gcode)
             if (path.config->isTravelPath())
                 speed *= extruder_plan.getTravelSpeedFactor();
             else
+            {
                 speed *= extruder_plan.getExtrudeSpeedFactor();
 
+                // for some movements such as prime tower purge, the speed may get changed by this factor
+                speed *= path.config->getExtrusionSpeedFactor();
```

Comment:
I think this edit is slightly in the wrong place.
It would be better 10 lines above.

Create a function `GCodePath::getSpeed()` which returns `config->getSpeed() * extrusion_speed_factor`

The line above should then just read `double speed = path.getSpeed();`